### PR TITLE
Add best practices section for rule 108

### PR DESCRIPTION
**Pull request by a1exsh**
Re: #792

### CODE REVIEW

File: chapters/http-requests.adoc [hunk 1]
```diff
@@ -193,6 +193,10 @@ suggest to choose one and only one of the following patterns per endpoint
 
 1. Use {PUT} with complete objects to update a resource as long as feasible
    (i.e. do not use {PATCH} at all).
+   *Note:* this choice by the API server imposes additional requirements on
+   the client (<<108>>). The <<appendix-best-practices>> section contains
+   recommendations on how to satisfy these requirements
+   (<<client-handling-compatible-extensions>>).
```
Comment (tfrauenstein) [inline chapters/http-requests.adoc:199]:
```suggestion
   *Note:* this choice by the API server imposes additional requirements on
   the client (<<108>>) which can be implemented e.g. in Java following the 
   practice <<handling-compatible-extensions>>.
```

File: chapters/compatibility.adoc [hunk 2]
```diff
@@ -133,6 +133,8 @@ Service clients should apply the robustness principle:
 ** Follow the redirect when the server returns HTTP status code {301} (Moved
   Permanently).
 
+The <<appendix-best-practices>> section contains recommendations on how to
+satisfy these requirements (<<client-handling-compatible-extensions>>).
```
Comment (tfrauenstein) [inline chapters/compatibility.adoc:137]:
```suggestion
The <<handling-compatible-extensions>> section describes a best practice to implement the requirements in Java. 
```

File: chapters/best-practices.adoc [hunk 3]
```diff
@@ -322,3 +322,25 @@ later than the given date:
...
+
+[[handling-compatible-extensions]]
+== Handling compatible API extensions
+
+[[client-handling-compatible-extensions]]
+=== In the clients
+
+Client code needs to be prepared to store any new optional properties of the
+fetched resource payload, and to serialize them later when submitting the
+payload back to the API server (<<108>>).
```
Comment (tfrauenstein) [inline chapters/best-practices.adoc:335]:
```suggestion
Client must not eliminate unknown optional fields from the 
fetched resource payload, and to serialize them later when submitting the
complete resource payload back to the API server via {PUT} (<<108>>).
```

**Review by tfrauenstein — COMMENTED** [review]
LTGM -- thank you!
Provided editorial suggestions.

**Comment by tkrop** [comment]
👍

**Comment by tfrauenstein** [comment]
👍

**Pull request merged**